### PR TITLE
Replace custom Enum with enum.Enum in nupic.frameworks.opf.opfutils

### LINKED ISSUE #2121: Convert nupic.frameworks.opf.opfutils.InferenceElement to enum.Enum
See parent task #2120

### PR DESCRIPTION
**Pull request by scottpurdy**
fixes #2121 

@oxtopus please review

### CODE REVIEW

File: nupic/frameworks/opf/opfutils.py [hunk 1]
```diff
@@ -71,16 +68,19 @@ def getInputElement(inferenceElement):
...
+    """ Returns True if the inference from this timestep predicteds the input
     for the NEXT timestep.
 
     NOTE: This should only be checked IF THE MODEL'S INFERENCE TYPE IS ALSO
-    TEMPORAL. That is, a temporal model CAN have non-temporal inference elements,
-    but a non-temporal model CANNOT have temporal inference elements
+    TEMPORAL. That is, a temporal model CAN have non-temporal inference
+    elements, but a non-temporal model CANNOT have temporal inference elements.
+
+    :param inferenceElement: the inference element
```
Comment (breznak) [inline nupic/frameworks/opf/opfutils.py]: nit: isn't this `@param`?

**Comment by breznak** [comment]
:+1:

**Comment by breznak** [comment]
looks like typo at: 
https://travis-ci.org/numenta/nupic/jobs/62447603#L4909

**Comment by scottpurdy** [comment]
@chetan51 please review

File: nupic/frameworks/opf/exp_generator/ExpGenerator.py [hunk 2]
```diff
@@ -1791,7 +1791,7 @@ def _generateMetricSpecs(options):
 
   # -----------------------------------------------------------------------
   # Metrics for classification
-  elif inferenceType in (InferenceType.TemporalClassification):
+  elif inferenceType in (InferenceType.TemporalClassification,):
```
Comment (chetan51) [inline nupic/frameworks/opf/exp_generator/ExpGenerator.py]: Why not use `inferenceType is InferenceType.TemporalClassification` here?

File: nupic/frameworks/opf/opfutils.py [hunk 3]
```diff
@@ -31,58 +32,57 @@
 from collections import namedtuple
 
 import nupic.data.jsonhelpers as jsonhelpers
-from nupic.support.enum import Enum
 
+g_temporalInferenceElements = None
```
Comment (chetan51) [inline nupic/frameworks/opf/opfutils.py]: Is it possible to avoid using globals?

**Pull request closed without merging**